File: project_sprint2_fe/sprint2-project-fe/src/service/cartService.js
import axios from "axios";

const URL_CART = "http://localhost:8080/api/member/cart/"

export const getCartByUserId = async (id) => {
  try {
      return await axios.get(URL_CART+id);
  }catch (e) {
      console.log(e);
  }
}

export const getCardByProductIdAndUserId = async (productId,userId) => {
  try {
      return await axios.get(URL_CART+productId+"/"+userId);
  }catch (e) {
      console.log(e);
  }
}

export const addToCart = async (cartDTO) =>{
    try {
        return await axios.post(URL_CART,cartDTO)
    }catch (e) {
        console.log(e)
    }
}

export const deleteToCart = async (id) =>{
    try {
        return await axios.delete(URL_CART+id)
    }catch (e) {
        console.log(e)
    }
}

export const getCartByCartId = async (id) =>{
    try {
        return await axios.get(URL_CART + "detail/" + id)
    }catch (e) {
        console.log(e)
    }
}
export const updateQuantityCart = async (id,cartDTO) => {
    try {
        return await axios.patch(URL_CART + "update-quantity/"+id,cartDTO)
    }catch (e) {
        console.log(e)
    }
}

File: project_sprint2/src/main/java/com/example/project_sprint2/service/cart/CartService.java
package com.example.project_sprint2.service.cart;

import com.example.project_sprint2.dto.CartDTO;
import com.example.project_sprint2.dto.ICartDetailDTO;
import com.example.project_sprint2.model.Cart;
import com.example.project_sprint2.repository.cart.ICartRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class CartService implements ICartService {

    @Autowired
    private ICartRepository cartRepository;
    @Override
    public List<ICartDetailDTO> getAllCartByUserId(int id) {
        return cartRepository.getCartByUser_Id(id);
    }

    @Override
    public Cart getCardByProductIdAndUserId(Integer productId, Integer userId) {
        return cartRepository.getCartByProduct_IdAndUser_Id(productId,userId);
    }

    @Override
    public void addToCart(CartDTO cartDTO) {
        cartRepository.addToCart(cartDTO.getQuantity(),cartDTO.getProductId(),cartDTO.getUserId());
    }

    @Override
    public void updateQuantityProduct(Integer quantity, Integer cartId) {
        cartRepository.updateQuantityProduct(quantity,cartId);
    }

    @Override
    public void deleteCartByCartId(Integer id) {
        cartRepository.deleteById(id);
    }

    @Override
    public Cart getCartByCartId(Integer id) {
        return cartRepository.getCartByCartId(id);
    }

    @Override
    public void deleteCartByUser_Id(Integer userId) {
        cartRepository.deleteCartByUser_Id(userId);
    }


}
